torch/_dynamo/convert_frame.py

```diff
@@ -915,7 +915,7 @@ def format_guard_failures() -> str:
             guarded_code = compile_inner(code, one_graph, hooks, transform)
             return guarded_code
         except Exception as e:
-            fail_type = str(type(e))
+            fail_type = type(e).__qualname__
             fail_reason = str(e)
```

Comment:
Aren't both of these lines dead code though? :thinking:

Reply:
They're not, because they're used in the `finally` block